r-package/policytree/src/tree_search.cpp

```diff
@@ -155,9 +157,13 @@ std::unique_ptr<Node> level_one_learning(const std::vector<flat_set>& sorted_set
       }
       auto next_value = it->get_value(p);
       split_counter += 1;
+      samples_counter += 1;
       if (value == next_value) {
         continue;
       }
+      if (samples_counter < min_node_size || num_points - samples_counter < min_node_size) {
+        continue;
```

Comment:
Just leaving a note here for future @kanodiaayush and @erikcs:

We could `break` instead of `continue` when the RHS of this condition is met, but leave it as this for simplicity.